frontend/src/scenes/session-recordings/mobile-replay/androidRecordingPromptBannerLogic.ts

```diff
@@ -0,0 +1,98 @@
+import { afterMount, kea, key, path, props, reducers, selectors } from 'kea'
+import { loaders } from 'kea-loaders'
+import { subscriptions } from 'kea-subscriptions'
+import api from 'lib/api'
+import posthog from 'posthog-js'
+
+import { HogQLQuery, NodeKind } from '~/queries/schema'
+
+import type { androidRecordingPromptBannerLogicType } from './androidRecordingPromptBannerLogicType'
+
+export interface AndroidRecordingPromptBannerLogicProps {
+    context: 'home' | 'events' | 'replay'
+}
+
+export type AndroidEventCount = {
+    version: string
+    count?: number
+}
+
+export const androidRecordingPromptBannerLogic = kea<androidRecordingPromptBannerLogicType>([
+    path(['scenes', 'session-recordings', 'SessionRecordings']),
+    key((props) => props.context),
+    props({} as AndroidRecordingPromptBannerLogicProps),
+    loaders(({ values }) => ({
+        androidVersions: [
+            [] as AndroidEventCount[],
+            {
+                loadAndroidLibVersions: async () => {
+                    if (values.androidVersions && values.androidVersions.length > 0) {
+                        // if we know they ever had android events, don't check again
+                        return values.androidVersions
+                    }
+
+                    const query: HogQLQuery = {
+                        kind: NodeKind.HogQLQuery,
+                        query: `SELECT properties.$lib_version AS lib_version,
```

Comment:
Should we use the materialized columns instead eg `mat_$lib` etc? not sure if this applies in this case.

Reply:
HogQL uses a materialised column wherever it can. This is true of existing insight code too... so it only really matters as a choice if we're executing sql directly against CH

Reply:
TIL, thanks